File: bcube-services/booking-service/src/main/java/com/bcube/bookingservice/service/dto/response/BookingDetailsResponse.java
package com.bcube.bookingservice.service.dto.response;

import com.bcube.bookingservice.persistance.entity.BookingStatus;
import com.bcube.bookingservice.service.dto.Classes.StudioDto;
import com.bcube.bookingservice.service.dto.Classes.UserDto;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.Instant;
import java.time.LocalDate;

@Data
@AllArgsConstructor
public class BookingDetailsResponse {
    private Long id;
    private UserDto user;
    private StudioDto studio;
    private LocalDate date;
    private Instant startTime;
    private Instant endTime;
    private BookingStatus status;

    private String accessCode;
}

File: bcube-web/src/app/models/responses/booking/booking-details-response.ts
import { BookingStatus } from "@models/booking-status.model";
import { Studio } from "@models/studio.model";
import { User } from "@models/user.model";

export interface BookingDetailsResponse {
    id: number;
    user: User;
    // Nullable defensively - see booking.model.ts.
    studio: Studio | null;
    date: string;
    startTime: string;
    endTime: string;
    status: BookingStatus;
    accessCode: string | null;
    clientSecret: string | null;
    amountDueCents: number | null;
}